dali/pipeline/operator/builtin/external_source_test.cc

```diff
@@ -22,9 +24,189 @@
 #include "dali/pipeline/executor/async_pipelined_executor.h"
 #include "dali/pipeline/operator/builtin/external_source.h"
 #include "dali/test/dali_test_config.h"
+#include "dali/c_api.h"
```

Comment:
I don't think we should be testing the C API here... These tests should be moved either to C API tests or to a separate translation unit (e.g. `external_source_c_test.cc`)

Reply:
I do not see any difference between these two approaches. I'd vote for keeping this way. @jantonguirao , what do you think?

Reply:
Cool. In that case I'll keep it inside `external_source_test.cc`